lib/app/analytics.coffee

```diff
@@ -0,0 +1,23 @@
+opts = require '../options'
+Analytics = require 'analytics-node'
+
+@setCampaign = (req, res, next) ->
+  return next() unless opts.SEGMENT_WRITE_KEY
+  req.session.modalId = req.params.modal_id
+  req.session.acquisitionInitiative = req.params.acquisition_initiative
```

Comment:
Curious why we have to set things to the session before Auth and the track call? Is it because if not we'll lose it after Auth?